lib/FusionInventory/Agent/Inventory.pm

```diff
@@ -478,6 +479,69 @@ sub saveLastState {
     my $tpp = XML::TreePP->new();
 }
 
+sub retrieveElementInSection {
+    my ($self, $sectionName, $fields) = @_;
+
+    my $section = $self->getSection($sectionName);
+    return unless $section;
+
+    my @elements = grep { my $hash = $_; _isKeyValueListInHash($hash, $fields) ? $hash : () } @$section;
+    return first { 1 } @elements;
```

Comment:
Usage of first() function is useless here, as the condition is always true. You'd better use either:
- return $elements[0]
- return shift @elements